### PR TITLE
Move the code for adding a new photo to the right area

### PR DESCRIPTION
**Pull request by orta**
Fixes a bug shipped in https://github.com/artsy/emission/pull/893 where the code is added to the wrong callback. I've added some tests, as this stuff is really tough to actually run in dev.

### CODE REVIEW

File: src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import ConsignmentBG from "../Components/ConsignmentBG"
 import ImageSelection, { ImageData } from "../Components/ImageSelection"
 import { BodyText as P } from "../Typography"
 
-import triggerCamera from "../../../NativeModules/triggerCamera"
+import { triggerCamera } from "lib/NativeModules/triggerCamera"
```
Comment (orta) [inline src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx]: we want to be moving away from default exports, and it was easier to write tests against as an individual export

File: src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx [hunk 2]
```diff
@@ -80,22 +80,7 @@ export default class SelectFromPhotoLibrary extends React.Component<Props, State
...
-          selection: this.state.selection,
-          // An item in cameraImage is a subset of the photo that we pass in,
-          // so we `as any` to avoid a compiler error
-          cameraImages: data.edges.map(e => e.node).concat(this.state.cameraImages as any),
-        })
-      } else {
-        console.log("CameraRoll: Did not receive a photo from call to getPhotos")
-      }
-    })
+    CameraRoll.getPhotos(fetchParams).then(data => this.appendAssets(data))
```
Comment (orta) [inline src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx]: I tried `CameraRoll.getPhotos(fetchParams).then(this.appendAssets)` but `this` gets all screwey.

File: src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx [hunk 3]
```diff
@@ -138,10 +123,18 @@ export default class SelectFromPhotoLibrary extends React.Component<Props, State
   }
 
   onPressNewPhoto = () => {
-    triggerCamera(this).then(photo => {
+    return triggerCamera(this).then(photo => {
```
Comment (orta) [inline src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx]: So you can write tests based on the async behavior

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
This was real tricky to actually validate - in the end I used the ability for the Emission beta to run against a PR, which meant I could ship a few guesses and download to test on device.

File: src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx [hunk 4]
```diff
@@ -137,11 +120,37 @@ export default class SelectFromPhotoLibrary extends React.Component<Props, State
     this.props.navigator.pop()
   }
 
+  getCameraRollPhotos = (params: any) => CameraRoll.getPhotos(params)
```
Comment (orta) [inline src/lib/Components/Consignments/Screens/SelectFromPhotoLibrary.tsx]: RN 0.48 did not include the Jest mapping for the camera, so by moving that call into this easily replaced function, it becomes easy to test

**Comment by sarahscott** [comment]
👍

**Pull request merged**